src/platform/src/main/java/org/geoserver/platform/resource/Files.java

```diff
@@ -44,7 +44,7 @@
         final File file;
 
         private ResourceAdaptor(File file) {
-            this.file = file;
+            this.file = file.getAbsoluteFile();
```

Comment:
When you to take the path of a resource, it isn't possible to derive from the path whether it is absolute or relative in Linux (Resource paths according to the specs you wrote do not support relative paths and make no difference between a starting slash and no starting slash). I don't see how this would be dangerous or affect anything. ResourceWrappers are used to provide simple files to objects and methods that expect a Resource. They should be threaten like a normal resource, and are mostly used to directly read and write to it. This change simply guarantees that the path returned is unique and makes some kind of consistent sense to somebody who is following the resource API.